apps/prairielearn/elements/pl-sketch/pl-sketch.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable */
+
+window.PLSketchApi = {
+  sketchpad: {},
+  color: '#000000',
+  tool: null,
+
+  changeTool: function (toolName) {
+    if (toolName === this.tool) {
+      return;
+    }
+    console.log(toolName);
+    
+    if (toolName === 'pen') {
+      size = 5;
+      this.tool = 'pen';
+      $('#pen').css('background', '#C0C0C0');
+      $('#eraser').css('background', '#FAFAFA');
+      this.sketchpad.penSize = size;
+      this.sketchpad.color = this.color;
+    } else {
+      size = 30;
+      this.tool = 'eraser';
+      $('#eraser').css('background', '#C0C0C0');
+      $('#pen').css('background', '#FAFAFA');
+      this.sketchpad.penSize = size;
+      this.sketchpad.color = '#FFF';
+    }
+  },
+
+  updateColor: function (colorName) {
+    this.color = colorName;
+    $('.pl-sketch-color-icon').css('backgroundColor', this.color);
+    if (this.tool === 'pen') {
+      this.sketchpad.color = this.color;
+    }
+  },
+
+  setupSketchpad: function (width, height, uuid) {
+    console.log('#undo-'+uuid);
+    let element = '#pl-sketch-canvas-' + uuid;
+    var sketchpadObject = {};
+    sketchpadObject['width'] = width;
+    sketchpadObject['height'] = height;
+    sketchpadObject['element'] = element;
+    let sketches_string = $('#pl-sketch-input-' + uuid).val();
+    console.log(sketches_string.length);
+    if (sketches_string.length != 0) {
+        sketches_string = sketches_string.replace(/'/g, '!@#$%');
+        sketches_string = sketches_string.replace(/"/g, '\'');
+        sketches_string = sketches_string.replace(/!@#\$%/g, '"');
+        let sketches = JSON.parse(sketches_string);
+        let strokes = sketches['sketches'];
+        strokes = strokes.replace(/'/g, '"');
+        strokes = JSON.parse(strokes);
+
+        sketchpadObject['strokes'] = strokes;
+        sketchpadObject['undoHistory'] = sketches['undoHistory'];
+    }
+    this.sketchpad = new Sketchpad(sketchpadObject);
```

Comment:
`this` in this case is `window.PLSketchAPI`, which is unique within the page. This will give you again issues with multiple sketches in the same question. I would encourage you take a look at the strategy used in other elements like [`pl-file-editor`](https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/elements/pl-file-editor/pl-file-editor.js) or [`pl-order-blocks`](https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.js) for ways to address this kind of process.